faithcomesbyhearing/biblebrain-domain
# FCBHDBP-595 biblebrain prep work

# Description
Move the files from fchb/biblebrain-common-old to biblebrain-domain and update the project to implement the hexagonal pattern.

## Screenshot
- The following is the new folder structure.
![Screenshot from 2023-08-14 10-19-17](https://github.com/faithcomesbyhearing/biblebrain-domain/assets/73488660/7b158ccf-a5ce-4f28-bc03-4fdc07f243f8)

## NOTE 1
The biblebrain-uploader project needs to use the GOPRIVATE env variable to use the common module because it is private module. Also, we need to change git config so that when downloadr tries to download the common module use `git@github.com` insteadOf `https://github.com/`
```shell
GOPRIVATE=github.com/faithcomesbyhearing
git config --global url."git@github.com:".insteadOf "https://github.com/"
```
## NOTE 2
It is related to bible-uploader PR:
- https://github.com/faithcomesbyhearing/biblebrain-uploader/pull/2

## Issue Link
https://fullstacklabs.atlassian.net/browse/FCBHDBP-595

## How Do I QA This
We need to import the biblebrain-common module into the biblebrain-uploader project and you should run the following command and it works correctly.

```bash
./bin/biblebrain-uploader timestamp-gap-fixer -l 3
```

```bash
# biblebrain-uploader is a command tool to:

Update records in the bible_file_timestamps entity using a specific SQL query

Usage:
  biblebrain-uploader timestamp-gap-fixer [flags]

Flags:
  -h, --help        help for timestamp-gap-fixer
  -l, --limit int   Limit of records to retrieve (default 10)
```
